ledger/simulation/simulator.go

```diff
@@ -197,12 +197,15 @@ func (s Simulator) evaluate(hdr bookkeeping.BlockHeader, stxns []transactions.Si
 	}
 
 	// Finally, process any pending end-of-block state changes.
-	vb, err := eval.GenerateBlock()
+	ub, err := eval.GenerateBlock(nil) // XXX not fetching proposer addresses
```

Comment:
XXX in non-test code is confusing.  Is this something that needs to be fixed?

Reply:
I was starting to learn toward some sort of `fakeAgreement` function for test code instead of stuff like this:
```
ledgercore.MakeValidatedBlock(ub.UnfinishedBlock(), ub.UnfinishedDeltas())
```
especially for the places that also fake a Seed and Proposer.
But I'm not sure what should be done in `simulate`. It feels a little wrong to supply an easy way to coerce an UnfinishedBlock to Proposable/Validated.  But then again, that's what we want to do.  As long as it's explicit, I suppose it's fine.